aspnetcore/tutorials/first-mvc-app/controller-methods-views.md

```diff
@@ -22,130 +23,121 @@ We have a good start to the movie app, but the presentation is not ideal. We don
 
-[Tag Helpers](../../mvc/views/tag-helpers/index.md) are one of the most popular new features in ASP.NET Core. See [Additional resources](#additional-resources) for more information.
-
-Open the `Movies` controller and examine the two `Edit` action methods:
+[Tag Helpers](xref:mvc/views/tag-helpers/intro) are one of the most popular new features in ASP.NET Core. See [Additional resources](#additional-resources) for more information.
 
-![Movies controller showing the Edit action method that takes an optional id parameter](controller-methods-views/_static/1.png)
+Open the `Movies` controller and examine the two `Edit` action methods. The following code shows the `HTTP GET Edit` method, which fetchs the movie and populates the edit form generated by the *Edit.cshtml* Razor file.
 
-[!code-csharp[Main](start-mvc/sample2/src/MvcMovie/Controllers/MoviesController.cs?range=78-92)]
+[!code-csharp[Main](start-mvc/sample/MvcMovie/Controllers/MC1.cs?name=snippet_edit1)]
 
-This method will be updated after we add a rating (add it to the bind).
+The following code shows the `HTTP POST Edit` method, which proccess the posted movie values:
 
-[!code-csharp[Main](start-mvc/sample2/src/MvcMovie/Controllers/MoviesController.cs?range=101-131)]
+[!code-csharp[Main](start-mvc/sample/MvcMovie/Controllers/MC1.cs?name=snippet_edit2)]
 
-The `[Bind]` attribute is one way to protect against [over-posting](http://www.asp.net/mvc/overview/getting-started/getting-started-with-ef-using-mvc/implementing-basic-crud-functionality-with-the-entity-framework-in-asp-net-mvc-application#overpost). You should only include properties in the `[Bind]` attribute that you want to change. See [Protect your controller from over-posting](http://www.asp.net/mvc/overview/getting-started/getting-started-with-ef-using-mvc/implementing-basic-crud-functionality-with-the-entity-framework-in-asp-net-mvc-application#overpost) for more information. [ViewModels](http://rachelappel.com/use-viewmodels-to-manage-data-amp-organize-code-in-asp-net-mvc-applications/) provide an alternative approach to prevent over-posting.
+The `[Bind]` attribute is one way to protect against [over-posting](https://docs.microsoft.com/aspnet/mvc/overview/getting-started/getting-started-with-ef-using-mvc/implementing-basic-crud-functionality-with-the-entity-framework-in-asp-net-mvc-application#overpost). You should only include properties in the `[Bind]` attribute that you want to change. See [Protect your controller from over-posting](http://www.asp.net/mvc/overview/getting-started/getting-started-with-ef-using-mvc/implementing-basic-crud-functionality-with-the-entity-framework-in-asp-net-mvc-application#overpost) for more information. [ViewModels](http://rachelappel.com/use-viewmodels-to-manage-data-amp-organize-code-in-asp-net-mvc-applications/) provide an alternative approach to prevent over-posting.
```

Comment:
data/ef-mvc/crud

It doesn't have overpost anchor right now, I'll add a commit to this branch to put it there.